main/src/ca/uwaterloo/flix/api/lsp/provider/completion/MagicMatchCompleter.scala

```diff
@@ -86,51 +84,55 @@ object MagicMatchCompleter {
   /**
     * Formats the cases of an enum into a string.
     */
-
-  private def generateCasesString(cases: Map[Symbol.CaseSym, TypedAst.Case]) = {
-   val maxCaseLength = cases.values.map(getCaseLength).max
-   cases.toList
-     .sortBy(_._1.loc)
-     .foldLeft(("", 1)) { case ((acc, z), (sym, cas)) =>
-       val (lhs, rhs, nextZ) = createCase(sym, cas, z)
-       val paddedLhs = padLhs(lhs, maxCaseLength)
-       (acc + s"    case $paddedLhs => $rhs\n", nextZ)
-     }._1
+  private def patternMatchBody(cases: Map[Symbol.CaseSym, TypedAst.Case]): String = {
+    val maxCaseLength = cases.values.map(getCaseLength).max
+    val sb = new StringBuilder
+    cases.toList
+      .sortBy(_._1.loc)
+      .foldLeft(1) { case (z, (sym, cas)) =>
```

Comment:
This longer has to be a fold right? It can even just be an ordinary for loop? I think

Reply:
The increment of z is not linear.
To use a for loop here, we'll need a mutable `var z` to track it. I've implemented it in the new commit. Shall we adopt this?

Reply:
Yeah, I think its good. Shall I merge?

Reply:
OK, let's merge.